template.yaml

```diff
@@ -250,15 +250,17 @@ Resources:
                   Action:
                     - ssm:SendCommand
                     - ssm:GetCommandInvocation
-                  Resource: '*'
+                  Resource:
+                    - !Sub "arn:aws:ssm:${AWS::Region}::document/AWS-RunShellScript"
                 - Effect: Allow
                   Action:
                     - ssm:SendCommand
                     - ssm:GetCommandInvocation
                   Resource: !Sub "arn:aws:ec2:${AWS::Region}:${AWS::AccountId}:instance/*"
                   Condition:
-                    StringLike:
-                      "ec2:ResourceTag/aws:autoscaling:groupName": !Sub "${AWS::StackName}-AgentAutoScaleGroup-*"
+                    StringEquals:
+                      "ssm:resourceTag/Role": "buildkite-agent"
+                      "ssm:resourceTag/aws:autoscaling:groupName": !Ref AgentAutoScaleGroup
```

Comment:
As above, this `Condition` is OK for `ssm:SendCommand`, but not for `ssm:GetCommandInvocation`. Even though it's annoyingly repetitive, I recommend separating the 2 Actions into distinct Statements.